### PR TITLE
KAADO-41:  Replace dummy user id with real user id in session

### PR DESCRIPTION
**Pull request by tsoibet**

### CODE REVIEW

File: src/services/userService.ts [hunk 1]
```diff
@@ -60,9 +62,11 @@ export async function getUser(
...
     try {
         await connectDB();
-        // TODO: Get USER_ID from session
-        const userId = stringToObjectId(userID);
-
+        const session = await getServerSession(authOptions);
+        if (!session) {
+            return { error: 'User not found' };
+        }
+        const userId = stringToObjectId(session.user.id);
```
Comment (nasudon) [inline src/services/userService.ts:67]: Functions in this service have the userID in parameter. You may either 
1. remove it completely or 
2. use it by removing the default value `= '123456789012' `, letting it be optional and updating the assignment
```
export async function getUser(
    userID: string?
): Promise<{ user?: UserBasicInfo; error?: string }> {
    try {
        await connectDB();
        let userId: mongoose.Types.ObjectId | null;
        if (userID) {
            userId = stringToObjectId(userID);
        } else {
            const session = await getServerSession(authOptions);
            if (!session) {
                return { error: 'User not found' };
            }
            userId = stringToObjectId(session.user.id);
        }
```

Comment (nasudon) [reply src/services/userService.ts:67]: The benefit to keep the `userID` is that it allows us to have the flexibility to administrate user in the future. If we don't have plan to do so, we can remove it.

**Review by nasudon — COMMENTED** [review]
Please check my comment

File: src/services/userService.ts (hunk 1 above)
Comment (tsoibet) [reply src/services/userService.ts:67]: Fixed by completely deleting it.

**Review by nasudon — APPROVED** [review]
LGTM

**Pull request merged**